src/charm.py

```diff
@@ -152,6 +162,10 @@ def __init__(self, *args: Any) -> None:
         )
         self.framework.observe(self.on.rotate_key_action, self._on_rotate_key_action)
 
+        self.framework.observe(
+            self.on[self._login_ui_relation_name].relation_changed, self._on_config_changed
```

Comment:
Let's not run a method that was used on config changed event, but just `_handle_status_update_config` instead